Core/Scripts: move npc_kharan_mighthammer to DB/SAI

tkrokli opened the pull request:
**Changes proposed:**

This PR affects and improves the gossip content of the following 2 quests:
- [Quest ID 4001 "What Is Going On?" - part 2 (Horde)](http://www.wowhead.com/quest=4001/what-is-going-on)
- [Quest ID 4342 "Kharan's Tale" (Alliance)](http://www.wowhead.com/quest=4342/kharans-tale)

Target of this PR:
Move core script `npc_kharan_mighthammer` to DB/SAI
(thereby removing 10 lines of hardcoded text).
-  add correct gossip flow for quest 4001 (all 10 lines)
-  use `conditions` `-OR-` to separate quest 4342 gossip flow from quest 4001
-  add missing `creature_text` to Kharan Mighthammer

**Target branch(es):** 3.3.5 **only** (quests removed in patch 4.0.3a)

**Issues addressed:** Updates https://github.com/velinath/TrinityCore/issues/2

**Tests performed:** (Does it build, tested in-game, etc)

Builds without issues (code removal only in core, SQL content added). Tested repeatedly with an Alliance and a Horde character to verify that the gossip flows correctly. Both quests complete as intended.
SQL part tested repeatedly by emptying the world DB and rebuilding it from worldserver startup.

**Known issues and TODO list:**
- [x] Gossip shows only "Greetings, $n." before quest taken and after quest complete
- [x] Redundant `SAY_GET_ME_OUT_OF_HERE` (core script) comment?

Commit:
Core/Scripts: move npc_kharan_mighthammer to DB/SAI

This PR affects the following 2 quests:
Quest ID 4001 "What Is Going On?" - part 2 (Horde)
Quest ID 4342 "Kharan's Tale" (Alliance)

Target of this PR:
Move core script npc_kharan_mighthammer to DB/SAI
(thereby removing 10 lines of hardcoded text).

offl commented on sql/updates/world/3.3.5/2016_08_08_4001_world_335.sql:
```diff
@@ -0,0 +1,56 @@
+
+DELETE FROM `creature_text` WHERE `entry` = @Kharan;
+INSERT INTO `creature_text` (`entry`,`groupid`,`id`,`text`,`type`,`language`,`probability`,`emote`,`duration`,`sound`,`BroadcastTextId`,`TextRange`,`comment`) VALUES
+(@Kharan,0,0,'HEY! HEY YOU! $R! Get me out of here!',12,0,100,0,0,0,4725,0,'Kharan Mighthammer - SAY_GET_ME_OUT_OF_HERE');
```
>SAY_GET_ME_OUT_OF_HERE

Guess this is not needed here.

tkrokli replied:
Fixed in the new commit.

Commit:
DB/conditions: extra gossip conditions

- added conditions for gossip before quest taken and after quest completed.
- removed redundant core script SAY_GET_ME_OUT_OF_HERE comment

offl commented on sql/updates/world/3.3.5/2016_08_08_4001_world_335.sql:
```diff
@@ -0,0 +1,60 @@
+
+DELETE FROM `creature_text` WHERE `entry` = @Kharan;
+INSERT INTO `creature_text` (`entry`,`groupid`,`id`,`text`,`type`,`language`,`probability`,`emote`,`duration`,`sound`,`BroadcastTextId`,`TextRange`,`comment`) VALUES
+(@Kharan,0,0,'HEY! HEY YOU! $R! Get me out of here!',12,0,100,0,0,0,4725,0,'Kharan Mighthammer');
```
All texts. If you want to add all.
(@Kharan,0,0,'Key... get the key... Gerstahn has... key.',12,0,100,0,0,0,4723,0,'Kharan Mighthammer'),
(@Kharan,0,1,'Try and make yourself useful, $r. GET ME OUT OF HERE! The High Interrogator has the key.',12,0,100,0,0,0,4724,0,'Kharan Mighthammer'),
(@Kharan,0,2,'HEY! HEY YOU! $R! Get me out of here!',12,0,100,0,0,0,4725,0,'Kharan Mighthammer'),
(@Kharan,0,3,'%s groans.',16,0,100,0,0,0,4726,0,'Kharan Mighthammer');

tkrokli replied:
Oh. OK, right. I noticed that those texts were located near each other in `broadcast_text`, but I thought it was because they were used by the other nearby prisoners. Very well, I will get to it and create a new commit for it. Thanks for the suggestion!

offl replied:
Marshal Windsor uses another texts, Commander Gor'shak too. Common prisoners(Slave, Tortured Slave) uses(not implemented) texts with BroadcastTextID 5458 - 5460, I guess.

Commit:
DB/creature: add more text to Kharan

- add all missing creature_text lines to entry 9021 Kharan Mighthammer
- add SAI duration to wait before TEXT_OVER is triggered (3000 ms)
